kyuubi-events/src/main/scala/org/apache/kyuubi/events/handler/EventHandlerLoader.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kyuubi.events.handler
+
+import java.util.ServiceLoader
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable.ArrayBuffer
+
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.events.KyuubiEvent
+
+object EventHandlerLoader {
+
+  def loadCustom(kyuubiConf: KyuubiConf): List[EventHandler[KyuubiEvent]] = {
+    val providers = ArrayBuffer[CustomEventHandlerProvider]()
+    ServiceLoader.load(
+      classOf[CustomEventHandlerProvider],
+      Thread.currentThread().getContextClassLoader)
+      .iterator()
+      .asScala
+      .foreach(provider => providers += provider)
+
+    providers.map(_.create(kyuubiConf)).toList
```

Comment:
what happens if it throw exception

Reply:
It will continue to throw it to the method of loadEventhandler. should we catch it?

Reply:
according to the listener extension of Spark, it should be better not affect the engine if the plugin fails to initialize. we can catch it and add some logging.

Reply:
catch it and add a test.